src/esp/sensor/VisualSensor.cpp

```diff
@@ -136,5 +137,39 @@ bool VisualSensor::getObservation(sim::Simulator& sim, Observation& obs) {
   return true;
 }
 
+void VisualSensor::moveSemanticSensorToSemanticSceneGraph(sim::Simulator& sim) {
+  CORRADE_INTERNAL_ASSERT(visualSensorSpec_->sensorType ==
+                          SensorType::Semantic);
+  // check if the sensor is already in this semantic scene graph
+  if (node().parent() ==
+      sim.getActiveSemanticSceneGraph().getRootNode().scene()) {
+    return;
+  }
+  // back up
+  relativeTransformBackup_ = node().transformation();
+  Mn::Matrix4 absTransform = node().absoluteTransformation();
+  semanticSensorParentNodeBackup_ =
+      static_cast<scene::SceneNode*>(node().parent());
+
+  // now, take the sensor from the current scene graph and connect it to
+  // the root node of semantic scene graph, set the *correct* transformation
+  node().setParent(&sim.getActiveSemanticSceneGraph().getRootNode());
+  node().setTransformation(absTransform);
+}
+
+void VisualSensor::moveSemanticSensorBackToRegularSceneGraph(
+    sim::Simulator& sim) {
+  CORRADE_INTERNAL_ASSERT(visualSensorSpec_->sensorType ==
+                          SensorType::Semantic);
+
+  // check if the sensor is already in this semantic scene graph
+  if (node().scene() == sim.getActiveSceneGraph().getRootNode().scene()) {
+    return;
+  }
+
+  node().setParent(semanticSensorParentNodeBackup_);
```

Comment:
Can you add an assert that the pointer is not null and the optional is not empty and then null/empty them after?

Reply:
OK. Will do.